File: backend/src/main/java/com/elbuensabor/entities/Factura.java
package com.elbuensabor.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "factura")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Factura extends Base {
    @Column
    private Date fecha;
    @Column
    private Integer numero;
    @Column
    private Double montoDescuento;
    @Column
    private String formaPago;
    @Column
    private String nroTarjeta;
    @Column
    private Double totalVenta;
    @Column
    private Double totalCosto;

    private Pedido pedido;
    private List<DetalleFactura> detallesFactura = new ArrayList<DetalleFactura>();
}

File: frontend/el-buen-sabor-frontend/src/app/entidades/Factura.ts
import { DetalleFactura } from './DetalleFactura';
import { Pedido } from './Pedido';

export class Factura {
  id: number = 0;
  fecha: Date = new Date();
  numero: number = 0;
  montoDescuento: number = 0;
  formaPago: string = '';
  nroTarjeta: string = '';
  totalVenta: number = 0;
  totalCosto: number = 0;
  pedidos: Pedido = new Pedido();
  detallesFactura: DetalleFactura[] = [];
}
